src/utils/index.ts

```diff
@@ -557,6 +557,13 @@ export const getEmojiMapAll = (emojiContainer: EmojiContainer): Map<string, Emoj
   });
   return emojiMap;
 };
+export const getEmojiListByCategoryIds = (emojiContainer: EmojiContainer, categoryIds: number[]): Array<Emoji> => {
+  if (!categoryIds) return getEmojiListAll(emojiContainer);
+
+  return emojiContainer?.emojiCategories?.filter((emojiCategory: EmojiCategory) => categoryIds.includes(emojiCategory.id))
+    .map((emojiCategory: EmojiCategory) => emojiCategory.emojis)
+    .reduce((prevArr: Array<Emoji>, currArr: Array<Emoji>) => prevArr.concat(currArr), []);
+};
```

Comment:
Probably we can simplify this like below
```
export const getEmojiListByCategoryIds = (emojiContainer: EmojiContainer, categoryIds: number[]): Array<Emoji> => {
  if (!categoryIds) return getEmojiListAll(emojiContainer);

  return emojiContainer?.emojiCategories
    ?.filter((emojiCategory: EmojiCategory) => categoryIds.includes(emojiCategory.id))
    .flatMap((emojiCategory: EmojiCategory) => emojiCategory.emojis) || [];
};
```

Reply:
Applied